airbyte-integrations/connectors/source-lexoffice/source_lexware/source.py

```diff
@@ -0,0 +1,247 @@
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.core import IncrementalMixin
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
+
+
+# Basic full refresh stream
+class LexwareStream(HttpStream, ABC):
+    """
+    This class represents a stream output by the connector.
+    This is an abstract base class meant to contain all the common functionality at the API level e.g: the API base URL, pagination strategy,
+    parsing responses etc..
+
+    Each stream should extend this class (or another abstract subclass of it) to specify behavior unique to that stream.
+
+    Typically for REST APIs each stream corresponds to a resource in the API. For example if the API
+    contains the endpoints
+        - GET v1/customers
+        - GET v1/employees
+
+    then you should have three classes:
+    `class LexwareStream(HttpStream, ABC)` which is the current class
+    `class Customers(LexwareStream)` contains behavior to pull data for customers using v1/customers
+    `class Employees(LexwareStream)` contains behavior to pull data for employees using v1/employees
+
+    If some streams implement incremental sync, it is typical to create another class
+    `class IncrementalLexwareStream((LexwareStream), ABC)` then have concrete stream implementations extend it. An example
+    is provided below.
+
+    See the reference docs for the full list of configurable options.
+    """
+
+    # url base. Required.
+    url_base = "https://api.lexoffice.io/v1/"
+
+    def __init__(self, authenticator: TokenAuthenticator = None, **kwargs):
+        super().__init__(authenticator=authenticator or None)
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        This method should return a Mapping (e.g: dict) containing whatever information required to make paginated requests. This dict is passed
+        to most other methods in this class to help you form headers, request bodies, query params, etc..
+
+        For example, if the API accepts a 'page' parameter to determine which page of the result to return, and a response from the API contains a
+        'page' number, then this method should probably return a dict {'page': response.json()['page'] + 1} to increment the page count by 1.
+        The request_params method should then read the input next_page_token and set the 'page' param to next_page_token['page'].
+
+        :param response: the most recent response from the API
+        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
+                If there are no more pages in the result, return None.
+        """
+        decoded_response = response.json()
+        if not bool(decoded_response.get("last", True)):
+            last_page = decoded_response.get("number", 0) + 1
```

Comment:
```suggestion
            last_page = decoded_response.get("number") + 1
```
It is better to make it failed if the number doesn't exist.